base/src/main/java/io/spine/validate/rule/ValidationRules.java

```diff
@@ -21,22 +21,33 @@
 package io.spine.validate.rule;
 
 import com.google.common.base.Splitter;
-import com.google.common.collect.ImmutableCollection;
 import com.google.common.collect.ImmutableSet;
-import io.spine.io.PropertyFiles;
-
+import com.google.protobuf.DescriptorProtos;
+import io.spine.annotation.Internal;
+import io.spine.logging.Logging;
+import io.spine.type.KnownTypes;
+import io.spine.type.MessageType;
+import org.slf4j.Logger;
+
+import java.io.Serializable;
 import java.util.Collection;
-import java.util.Properties;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static com.google.common.collect.Sets.newHashSetWithExpectedSize;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
 
 /**
  * A collection of {@linkplain ValidationRule validation rules} known to the application.
  *
  * <p>During initialization of this class, definitions of validation rules are verified.
  * If an invalid validation rule was found, a runtime exception will be thrown.
  */
-public final class ValidationRules {
+public final class ValidationRules implements Serializable {
```

Comment:
Are these external message constraints?

For a while, we have been referring to the "validation rules" as to the "message constraints. And as I understand, this class is about `ExternalMessageConstraints`.